### PR TITLE
feat: Add support for list of capabilities in samconfig.toml

### PR DESCRIPTION
**Pull request by G4brym**
#### Which issue(s) does this change fix?
#2417 

#### Why is this change necessary?
To allow for capabilities to be configured in a list orientation, while maintaining support for the previous configurations

#### How does it address the issue?
It adds support for the `tomlkit.items.Array` class
I decided to import the `tomlkit.items.Array` as `tomlkit_Array` to not cause confusion with python's array

#### What side effects does this change have?
None, that i found

#### Checklist
- [ ] Write documentation

This is my first pr, and i think the change is so small that the only item in the checklist that applies here is the Documentation, but i didn't find the write place to do it.

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by moelasmar — CHANGES_REQUESTED** [review]
Could you add some uint/integration test cases that cover this change.

**Comment by G4brym** [comment]
Just added some unit tests for my changes.
After digging more into the tomlkit, I found that the `tomlkit.items.Array` behaves like a normal list and the result of the `isinstance(value, list)` is also true when using the `tomlkit.items.Array` class.
So, I modified the code to use a list instead.

File: samcli/commands/_utils/options.py [hunk 1]
```diff
@@ -362,7 +362,7 @@ def notification_arns_override_option(f):
 
 
 def _space_separated_list_func_type(value):
-    return value.split(" ") if not isinstance(value, tuple) else value
+    return value.split(" ") if not isinstance(value, tuple) and not isinstance(value, list) else value
```
Comment (mgrandis) [inline samcli/commands/_utils/options.py:365]: Hi @G4brym, thank you for your contribution!

I think this code could be clearer:
```python
    if isinstance(value, str):
        return value.split(" ")
    if isinstance(value, (list, tuple)):
        return value
    raise ValueError()
```

What do you think?

**Comment by G4brym** [comment]
Thanks for pointing it out @mgrandis
I'm used to make the code more compact than it should, but doing it your way does make the code much more readable.
I'm looking forward to help you guys with more issues, I'm just picking some smaller first to know the project better :smile:

**Comment by mgrandis** [comment]
Thanks for the update @G4brym!
I see you added a test with an integer, could you also add ones for the other types that [TOML](https://toml.io) supports?
I think float, boolean, datetime, dict/object should be enough.

**Comment by G4brym** [comment]
Hey, I just added those data types you mentioned

**Review by mgrandis — APPROVED** [review]

**Comment by elbayaaa** [comment]
Do we need to make changes [here](https://github.com/aws/aws-sam-cli/blob/07fcd611d61eb2f213d03702391929ba22550adf/samcli/commands/deploy/guided_config.py#L59-L60) as well? I'm afraid that this change will successfully read the capabilities array from the toml file while executing `sam deploy` then override it to a space-separated string while writing to the file again.

**Comment by benkehoe** [comment]
@elbayaaa I think it'd be preferable to have that use the list format as well.

**Comment by elbayaaa** [comment]
@benkehoe yes I agree on that. I'm just worried about the inconsistency with the other options [here](https://github.com/aws/aws-sam-cli/blob/07fcd611d61eb2f213d03702391929ba22550adf/samcli/commands/deploy/guided_config.py#L59-L60). I think the scope should be expanded to include all possible options not just the capabilities.

**Comment by mgrandis** [comment]
This could indeed be a good opportunity to use the same method (list) everywhere, we need to be cautious with regressions however.

**Review by elbayaaa — APPROVED** [review]
Approving with the understanding that this is going to be an ongoing improvement. The next step is to save the value as an array instead of space-separated string.

**Review by moelasmar — APPROVED** [review]

**Comment by mgrandis** [comment]
Thank you for your contribution @G4brym!

**Pull request merged**